### DISCUSSION TITLE
ipv6 disable

### DISCUSSION THREAD

**Discussion by pikachu937** [bluenviron/mediamtx#1385] [Questions]
Hello. Help me please. tell me how i can disable start rtsp-simple-server on ipv6 interface? i disable ipv6 in system, but i can see in netstat response:
tcp6       0      0 :::8554                :::*                    LISTEN      2758280/rtsp-simple
tcp6       0      0 :::8889                 :::*                    LISTEN      2758280/rtsp-simple
udp6       0      0 :::8000                :::*                                2758280/rtsp-simple
udp6       0      0 :::8001                :::*                                2758280/rtsp-simple

Thank you

**Comment by pikachu937** [comment]
Answer please

**Comment by aler9** [comment]
<s>Hello, edit the configuration file and use an explicit listen IP. For instance, if your local ipv4 is 192.168.6.4:

```yml
rtspAddress: 192.168.6.4:8554
webrtcAddress: 192.168.6.4:8889
...
```

The server will listen on the specific ipv4 only.</s>

**Comment by pikachu937** [comment]
Hello, i checked and this work, but not for me.
I use variable runOnReady where i set rtsp path as localhost(for example rtsp://localhost:$RTSP_PORT/$RTSP_PATH) and after set variable rtspAddress: XXX.XXX.XXX.XXX:8554 i can't connecting to the port on address localhost, i get error: ERROR:libav.tcp:Connection to tcp://localhost:38652?timeout=10000000 failed: Connection refused. This is logical, but I would like to be able to disable ipv6 and be able to work through the localhost interface.

**Reply by aler9** [reply]
<s>The mapping between localhost and an IPv6 address is performed by the operating system, not by the server. If you still have problems you can disable IPv6 at the OS level by following a guide like this:

https://www.techrepublic.com/article/how-to-disable-ipv6-on-linux/</s>

**Reply by pikachu937** [reply]
Thanks, that's what I did first. this i write in first message - " i disable ipv6 in system, but i can see in netstat response"

**Reply by pikachu937** [reply]
If you look at the simple openssh or other network services, they either have a setting in the config file to disable or work according to the system settings and if the system is configured not to use ipv6, then the service will not be advertised. Therefore, provided that ipv6 is disabled on the system, rtsp-simple-server should not advertise on the disabled interface.

> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether XX:XX:XX:XX:XX:XX brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet XXX.XXX.XXX.XXX/XX brd XXX.XXX.XXX.XXX scope global eth0
       valid_lft forever preferred_lft forever

>  sysctl -a | grep -E 'net.ipv6.conf.all.disable_ipv6|net.ipv6.conf.default.disable_ipv6|net.ipv6.conf.eth0.disable_ipv6'
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.eth0.disable_ipv6 = 1

> netstat -tulnp | grep :::
tcp6       0      0 :::8554                :::*                    LISTEN      3974316/rtsp-simple
udp6       0      0 :::8000                :::*                                3974316/rtsp-simple
udp6       0      0 :::8001                :::*                                3974316/rtsp-simple

and now answer please, why?

**Reply by Docteh** [reply]
If IPv6 is disabled at the OS level, why is a mere user program able to use it? Disabled as in no IPv6 addresses assigned to any interface but the protocol is still available?

**Reply by aler9** [reply]
i've checked again and realized that using an explicit ipv4 listen address is not enough to stop listening on ipv6. We need to add an option to listen on either ipv4, ipv6 or both. I've created a feature request in order to track the issue: #1665
